packages/tracing/src/transaction.ts

```diff
@@ -114,9 +120,29 @@ export class Transaction extends SpanClass implements TransactionInterface {
 
     if (hasMeasurements) {
       logger.log('[Measurements] Adding measurements to transaction', JSON.stringify(this._measurements, undefined, 2));
-      transaction.measurements = this._measurements;
+      transactionEvent.measurements = this._measurements;
     }
 
-    return this._hub.captureEvent(transaction);
+    transactionEvent.tracestate = this.tracestate;
+
+    return this._hub.captureEvent(transactionEvent);
+  }
+
+  /**
+   * Create a new tracestate header value
+   *
+   * @returns The new tracestate value, or undefined if there's no client or no dsn
+   */
+  private _getNewTracestate(): string | undefined {
+    const client = this._hub.getClient();
+    const dsn = client?.getDsn();
+
+    if (!client || !dsn) {
+      return;
+    }
+
+    const { environment, release } = client.getOptions() || {};
+
+    return computeTracestate({ trace_id: this.traceId, environment, release, public_key: dsn.user });
```

Comment:
NIT: no `user_segment` but I guess this will come at some point (maybe add a comment about it).

Reply:
@jan-auer and I agreed we were leaving this out for now, as it adds a whole 'nother layer of complication. (We'd need an API for letting the (dev-)user set it, we'd have to make sure it got attached before the transaction closed (what happens if looking up the user and slow and the transaction times out first?), it would involve modifying the `tracestate` value midstream, which we don't currently do...)

Also, we never (AFAIK) resolved the question of whether this specifically has to be about the (end-)user, or whether to open it up to any (dev-)user-provided value (since effectively all we're really doing by adding `user-segment` is letting them add a tag with a pre-chosen name).